smslibrary/src/main/java/com/eis0/smslibrary/SMSMessage.java

```diff
@@ -1,34 +1,75 @@
 package com.eis0.smslibrary;
 
+import androidx.annotation.NonNull;
+
+/**
+ * Class implementing the Message interface, it represent a message of type SMS.
+ *
+ * @author Marco Cognolato
+ */
 public class SMSMessage implements Message {
+
+    // Public needed for test, but is also good to let the user know
+    public static final int MAX_MESSAGE_LENGTH = 160;
+
     private SMSPeer destination;
     private String message;
 
     /**
-     * Builds and returns an SMS message given a valid SMSPeer and a valid message to send
+     * Builds and returns an SMS message given a valid SMSPeer and a valid message to send.
+     *
+     * @param destination The destination Peer object.
+     * @param message The String message.
+     * @throws IllegalArgumentException If the message body is empty or exceeds the MAX_MESSAGE_LENGTH.
+     * @author Marco Cognolato
      */
     public SMSMessage(SMSPeer destination, String message) {
         this.destination = destination;
         this.message = message;
+        if(!isValid())
+            throw new IllegalArgumentException("Cannot create SMSMessage: invalid message body.");
     }
 
     /**
-     * Returns the peer of the message if valid
+     * Returns the peer of the message if valid.
+     *
+     * @return An SMSPeer object containing all the peer info.
+     * @author Marco Cognolato
      */
+    @Override
     public SMSPeer getPeer() {
         return destination;
     }
 
     /**
-     * Returns the data of the message if valid
+     * Returns the data of the message if valid.
+     *
+     * @return A String containing the data.
+     * @author Marco Cognolato
      */
+    @Override
     public String getData() {
         return message;
     }
 
     /**
-     * Helper function to write the message as a string
+     * Tells if an SMSMessage object is valid.
+     *
+     * @return True if the object is valid, false otherwise.
+     * @author Matteo Carnelos
+     */
+    public boolean isValid() {
+        return !message.isEmpty() && message.length() <= MAX_MESSAGE_LENGTH;
+    }
+
+    /**
+     * Helper function to write the message as a string.
+     *
+     * @return The String containing the object representation.
+     * @author Marco Cognolato
      */
+    @NonNull
+    @Override
     public String toString() {
         return "SMSPeer: " + destination + ", SMSMessage: " + message;
     }
```

Comment:
E' sufficiente inserire **destination**? Perchè vedo che hai scritto un metodo apposito per gli oggetti di tipo SMSPeer .toString() che restituisce l'address dell'oggetto SMSPeer. Scrivendo solo **destination** cosa ti mostra?

Reply:
L'operatore "+" applicato tra oggetti String e oggetti non stringa invoca automaticamente il metodo 'Object.toString()' sull'oggetto non stringa. Di conseguenza, facendo l'override di `Object.toString()` in SMSPeer otteniamo che un SMSPeer, se coinvolto in operazioni tra stringhe, verrà visto come l'address dell'oggetto.

Reply:
Ah, molto comodo. Ecco perchè consigliavano di creare sempre o sovrascrivere toString().